lib/common/mainloop.c

```diff
@@ -1344,6 +1344,28 @@ drain_timeout_cb(gpointer user_data)
     return FALSE;
 }
 
+/*!
+ * \brief Drain some remaining main loop events then quit it
+ *
+ * \param[in] mloop  Main loop to drain and quit
+ * \param[in] n      Drain up to this many pending events
+ */
+void
+pcmk_quit_main_loop(GMainLoop *mloop, unsigned int n)
```

Comment:
Would it be useful to have this function drain all remaining events (can we even know that?) if n=0, or n=-1, or something?

Reply:
There's honestly not much point to doing this at all -- it's intended for command-line tools that are exiting, to do any memory clean-up in callbacks so that valgrind output is easier to look at. The idea of the limit is not to waste too much time on that in production. :) It would probably be a good idea to do some real-world testing to find out the ideal balance there, but it doesn't seem worth the time at the moment.

In any case it would be easy enough to add what you suggest without breaking backward compatibility, so we can leave it until we need it.